### DISCUSSION TITLE
Blazor WASM - CORS

### DISCUSSION THREAD

**Discussion by supertom44** [ServiceStack/Discuss#19] [Q&A] [answered]
Hey,

I've been developing a site using the Blazor wasm template and I am now trying to get it hosted.

I'm using the included docker compose scripts and deploying it to my own VPS using the included github actions.

I'm not hosting the ui on a cdn due to this [issue](https://github.com/actions/checkout/issues/760) so thought it would be easier to keep them together for now.

However no matter what configuration I try in the AppHost.Configure I'm always getting the following error:

`Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://<ip address>/api/Authenticate. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing).`

I have reverted back to the original configuration in the template but still getting the error:
`Plugins.Add(new CorsFeature(allowOriginWhitelist:new[]{ 
            "http://localhost:5000",
            "https://localhost:5001",
            "https://" + Environment.GetEnvironmentVariable("DEPLOY_CDN")
        }, allowCredentials:true));`

I've tried removing cors altogether seen as I'm not deploying to a cdn but I still get the issue.

Not sure what I'm missing, any help would be appreicated.

Many thanks.

Tom

**Comment by supertom44** [comment]
I've got it hosted on github pages now, but still getting the same error.

Looking at the example template I can see that the requests send across the following headers:
`access-control-allow-credentials: true
access-control-allow-headers: Content-Type,Authorization
access-control-allow-methods: GET, POST, PUT, DELETE, PATCH, OPTIONS, HEAD
access-control-allow-origin: https://blazor-wasm.jamstacks.net
access-control-max-age: 600`

I can see that my site does not send these headers.
I can't figure out what's different between the template and my site.

**Comment by mythz** [comment]
If you're not deploying to a CDN then you should not be running into any CORS issue since the client and server would be deployed and hosted at the same site so there would be no need for CORS. 

FYI we've found a workaround of git's new fatal: unsafe repository issue by using a different action to deploy to GitHub Pages:
https://github.com/NetCoreTemplates/blazor-wasm/commit/31071f6e1139aedccb679612e3feafc3343a24f1

**Reply by supertom44** [reply]
I've tried enabling CORS using this [article](https://enable-cors.org/server_nginx.html) and placing it in the default_location, but still no joy.

When getting the demo site hosted did you have to make any other changes to the nginx configuration?

**Reply by supertom44** [reply]
That's what I thought which is why I couldn't figure out why it wasn't working when hosted on same box.

Running it locally the headers get set fine so my suspicion is it's something to do with the nginx proxy container.

**Reply by supertom44** [reply]
I've created a new project from the defaul template, not changed anything, deployed it to a new digital ocean droplet.
Not set a cdn so that it can server it all from the same box and I'm still getting a CORS error.
The hosted site is here: http://159.65.86.90/
Edit: Repo source code is here: https://github.com/supertom44/Blazor-SeviceStack-Test

**Reply by supertom44** [reply]
I have a feeling it's actually the fact that it doesn't have a domain so lets encrypt can't create a certificate for it. 
Just in the process of setting a domain up to test the theory.

**Comment by supertom44** [comment, marked as answer]
Turns out the CORS error was actually hiding the SSL error.
Set up a domain, rebuilt the nginx container and all is working now.
